File: packages/http-client-java/generator/http-client-generator-test/src/test/java/parameters/query/QueryTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package parameters.query;

import org.junit.jupiter.api.Test;

public class QueryTests {

    private final ConstantClient constantClient = new QueryClientBuilder().buildConstantClient();
    private final SpecialCharClient specialCharClient = new QueryClientBuilder().buildSpecialCharClient();

    @Test
    public void testConstant() {
        constantClient.post();
    }

    @Test
    public void testDollarSign() {
        specialCharClient.dollarSign("status eq 'active'");
    }
}

File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Parameters/Query/QueryTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.ClientModel;
using System.Threading.Tasks;
using NUnit.Framework;
using Parameters.Query;

namespace TestProjects.Spector.Tests.Http.Parameters.Query
{
    public class QueryTests : SpectorTestBase
    {
        [SpectorTest]
        public Task Constant() => Test(async (host) =>
        {
            ClientResult result = await new QueryClient(host, null).GetConstantClient().PostAsync();
            Assert.AreEqual(204, result.GetRawResponse().Status);
        });
    }
}
